### PR TITLE
feat: Increase robustness of Python setup script

### PR DESCRIPTION
**Pull request by stephenswat**
We've previously run into some minor issues with the Python setup script. This is only natural because shell scripting is hell, but we can still try to make it a little more robust. To this end, this commit adds two new features to the setup script:

1. The setup script now supports ZSH as well as Bash, so it should also work on Mac operating systems.
2. The script now tells the user which version of Python the bindings were set up for, so they don't run into versioning conflicts.

### CODE REVIEW

**Comment by stephenswat** [comment]
Wonderful, it looks like the Github runners are having some trouble. :cry:

**Comment by paulgessinger** [comment]
It's failing to download G4 data files from CERN. Should hopefully be transient.

File: Examples/Python/setup.sh.in [hunk 1]
```diff
@@ -1,3 +1,28 @@
-python_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}"  )" &> /dev/null && pwd  )
+# This file is part of the Acts project.
+#
+# Copyright (C) 2016-2020 CERN for the benefit of the Acts project
```
Comment (paulgessinger) [inline Examples/Python/setup.sh.in:3]: If you put the statement, you could update the year.

Comment (stephenswat) [reply Examples/Python/setup.sh.in:3]: Good point :thinking:

File: Examples/Python/setup.sh.in [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+
+if [ -n "$ZSH_VERSION" ]; then
+    export PYTHONPATH=${0:a:h}:$PYTHONPATH
+elif [ -n "$BASH_VERSION" ]; then
+    python_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}"  )" &> /dev/null && pwd  )
+    export PYTHONPATH=$python_dir:$PYTHONPATH
+else
+    # If the current shell is not ZSH or Bash, we can't guarantee that the
+    # script will work, so we throw an error.
+    echo "ERROR:   neither ZSH or Bash was detected, other shells are not supported. The environment has not been modified."
```
Comment (AJPfleger) [inline Examples/Python/setup.sh.in:20]: shouldn't it be "neither ... nor"?

**Review by AJPfleger — COMMENTED** [review]
Changes look good to me. The suggested change in the ERROR-message is imho very minor and may be ignored.

File: Examples/Python/setup.sh.in (hunk 2 above)
Comment (stephenswat) [reply Examples/Python/setup.sh.in]: Good point! I've fixed it. :smile:

**Review by paulgessinger — APPROVED** [review]
:+1:

**Pull request merged**